### PR TITLE
HSEARCH-2433 AssertionFailure with providedId in embeddeds

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-2433

Details in the commit comments.

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/metadata/impl/AnnotationMetadataProvider.java [hunk 1]
```diff
@@ -427,37 +430,57 @@ private Annotation getIdAnnotation(XProperty member, TypeMetadata.Builder typeMe
...
+		}
+
+		DocumentFieldMetadata fieldMetadata = fieldMetadataBuilder.build();
 
 		propertyMetadataBuilder.addDocumentField( fieldMetadata );
 		propertyMetadata = propertyMetadataBuilder.build();
-		typeMetadataBuilder.idProperty( propertyMetadata );
+
+		if ( isRoot ) {
+			typeMetadataBuilder.idProperty( propertyMetadata );
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/search/engine/metadata/impl/AnnotationMetadataProvider.java]: Mmmmh, did I miss something or if `!isRoot`, the whole work we did above was totally useless?

**Comment by Sanne** [comment]
@yrodiere you mention on the JIRA comments that this problem affects the Lucene backend as well.

Worth moving the test to `-engine` ?

**Comment by yrodiere** [comment]
@Sanne Yes, I'll see if I can highlight the issue in an -engine test.

**Comment by Sanne** [comment]
I meant to suggest to move this one, not to add an additional one. We don't want to have too many redundant tests checking for the same things ;)

**Comment by yrodiere** [comment]
@Sanne I'll move to -orm, then. -engine tests are not yet executed on Elasticsearch. Not even sure yet when they will be...

**Comment by Sanne** [comment]
> I'll move to -orm, then. -engine tests are not yet executed on Elasticsearch. 

Right, forgot about that. Let's merge this then, the missing test for the embedded mode is an orthogonal issue.

**Comment by yrodiere** [comment]
@Sanne I rebased on master: the build was failing because of https://hibernate.atlassian.net/browse/HSEARCH-2438 , which was merged to master a few minutes ago

**Comment by yrodiere** [comment]
@gsmet I addressed your concerns. Thanks!

**Comment by Sanne** [comment]
merged, thanks all!

**Pull request closed without merging**